test(e2e) : add tests for website learning

achalbajpai opened the pull request:
### What 
PR address #584 sub-point `Website learning (UI smoke test only)`

### Playwright Test Report 
<img width="1765" height="845" alt="image" src="https://github.com/user-attachments/assets/0358945f-65a1-4e02-bf70-13ca49810061" />

### Local Tests
<img width="587" height="162" alt="image" src="https://github.com/user-attachments/assets/2e9c5769-a724-43a1-8fef-8e8ae566b673" />



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Added comprehensive end-to-end tests for the Website Learning UI, including form interactions, validation, and website addition flows.
  * Introduced a new page object to streamline and standardize test interactions with the Website Learning feature.
* **Chores**
  * Improved test reliability by adding a test identifier to website list items.
  * Enhanced toast notification detection with a new utility function and refactored related test code.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

achalbajpai commented on tests/e2e/utils/page-objects/websiteLearningPage.ts:
```diff
@@ -0,0 +1,113 @@
+    }
+    
+    if (!found) {
+      const fallbackToast = this.page.locator(`text=${message}`).first();
+      await expect(fallbackToast).toBeVisible({ timeout: 8000 });
```
I don't think this is needed in our case

binary-koan commented on tests/e2e/website-learning/websiteLearning.spec.ts:
```diff
@@ -0,0 +1,54 @@
+    }
+  });
+
+  test("displays main website learning interface", async () => {
+    await websiteLearningPage.expectWebsiteLearningSection();
+    await websiteLearningPage.expectAddWebsiteButton();
+  });
+
+  test("shows add website form on button click", async () => {
+    await websiteLearningPage.clickAddWebsite();
+    await websiteLearningPage.expectAddWebsiteForm();
+  });
```
I think we can merge these

```suggestion
  test("displays the website learning section and add website form", async () => {
    await websiteLearningPage.expectWebsiteLearningSection();
    await websiteLearningPage.expectAddWebsiteButton();

    await websiteLearningPage.clickAddWebsite();
    await websiteLearningPage.expectAddWebsiteForm();
  });
```

binary-koan commented on tests/e2e/utils/page-objects/websiteLearningPage.ts:
```diff
@@ -0,0 +1,113 @@
+  private readonly urlLabel = 'label[for="url"]';
+  private readonly cancelButton = 'button:has-text("Cancel")';
+  private readonly submitButton = 'form button[type="submit"]';
+  private readonly errorText = '.text-sm.text-destructive';
+  private readonly websiteItems = 'div.rounded-lg.border';
```
I think we can improve these a bit

* For the error text we don't care about where it is and what class name it has, so I think we can just check the page contains the expected error text anywhere
* For the website items I think it's worth adding a `data-testid`, this class is too generic and doesn't indicate what the element actually is

binary-koan commented on tests/e2e/utils/page-objects/websiteLearningPage.ts:
```diff
@@ -0,0 +1,113 @@
+    return { name, url };
+  }
+
+  async expectToastMessage(message: string) {
```
Can we extract this to somewhere shared? (I see we have a similar `waitForToast` function in the saved replies page, and I expect we'll want it in a bunch of other places too)

binary-koan reviewed (commented):
Thanks! Added a few comments

achalbajpai commented:
> Thanks! Added a few comments

Thanks for the review . I have addressed all the comments mentioned. I hope the PR is good to be merged, I’m happy to fix any other issues if there are any left.

<img width="1117" height="560" alt="image" src="https://github.com/user-attachments/assets/9b80ed6a-7bbc-49e3-899d-302b4d7a842b" />

binary-koan commented on tests/e2e/utils/toastHelpers.ts:
```diff
@@ -0,0 +1,23 @@
+    '[aria-live="polite"]'
+  ];
+  
+  for (const selector of toastSelectors) {
```
We need to keep the `toastFound` logic here, otherwise this may be super slow while it waits for all the possible selectors. Alternatively we could just use a single selector instead of an array - all our toasts are the same so I'm not sure why we need to test a bunch of different selectors 🤔

achalbajpai replied:
> We need to keep the `toastFound` logic here, otherwise this may be super slow while it waits for all the possible selectors. Alternatively we could just use a single selector instead of an array - all our toasts are the same so I'm not sure why we need to test a bunch of different selectors 🤔

Added back the toastFound function; it does improve the speed. Also, since all the selectors were finding the same element, it’s better to keep just one , rather than covering all scenerios.

binary-koan reviewed (commented):
Added one more comment, looks good apart from that!

achalbajpai commented:
> Added one more comment, looks good apart from that!

Addressed the issues mentioned @binary-koan  , I hope all the changes are good to go.

binary-koan reviewed (approved):
LGTM!
